### PR TITLE
refactor!: clean up the database model

### PR DESCRIPTION
**Pull request by wgwz**
## Description

Closes: https://github.com/regen-network/regen-registry/issues/1551

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] targeted `dev` branch
- [x] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] submit a PR against `master` branch after this one (and other PRs depending on this one, e.g. on regen-network/regen-web, if applicable) get merged

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: sql/V3_01__db_cleanup.sql [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ALTER TABLE IF EXISTS credit_vintage DROP COLUMN IF EXISTS initial_distribution;
+ALTER TABLE IF EXISTS credit_vintage DROP COLUMN IF EXISTS issuer_id;
+ALTER TABLE IF EXISTS credit_vintage DROP COLUMN IF EXISTS methodology_version_created_at;
+ALTER TABLE IF EXISTS credit_vintage DROP COLUMN IF EXISTS methodology_version_id;
+ALTER TABLE IF EXISTS credit_vintage DROP COLUMN IF EXISTS reseller_id;
+ALTER TABLE IF EXISTS credit_vintage DROP COLUMN IF EXISTS start_date;
+ALTER TABLE IF EXISTS credit_vintage DROP COLUMN IF EXISTS tokenizer_id;
+ALTER TABLE IF EXISTS credit_vintage DROP COLUMN IF EXISTS tx_hash;
+ALTER TABLE document DROP COLUMN IF EXISTS event_id;
+ALTER TABLE organization DROP COLUMN IF EXISTS type;
```
Comment (blushi) [inline sql/V3_01__db_cleanup.sql:31]: given https://github.com/regen-network/registry-server/pull/311/files, I think we could also drop `website` from `organization` (making sure to migrate any existing data to `party.website_link` first)

**Comment by blushi** [comment]
@wgwz could you address the suggested changes above? ty!

**Review by blushi — APPROVED** [review]
Nice work, let's only merge as soon as https://github.com/regen-network/regen-registry/issues/1665 is addressed

File: sql/V3_01__db_cleanup.sql [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+DROP POLICY IF EXISTS credit_vintage_select_all ON credit_vintage;
+ALTER TABLE IF EXISTS credit_vintage DISABLE ROW LEVEL SECURITY;
+DROP POLICY IF EXISTS organization_insert_admin ON organization;
+DROP POLICY IF EXISTS organization_update_admin ON organization;
+DROP POLICY IF EXISTS owner_update_organization ON organization;
+DROP POLICY IF EXISTS organization_insert ON organization;
+DROP POLICY IF EXISTS organization_select_all ON organization;
+ALTER TABLE IF EXISTS organization DISABLE ROW LEVEL SECURITY;
+ALTER TABLE credit_class DROP COLUMN IF EXISTS designer_id;
+ALTER TABLE credit_class DROP COLUMN IF EXISTS standard;
```
Comment (blushi) [inline sql/V3_01__db_cleanup.sql:11]: I think we can also drop `methodology_id` from `credit_class` table

Comment (wgwz) [reply sql/V3_01__db_cleanup.sql:11]: yup, good catch. i pushed a new commit for that, and also dropped `address_id` from `party`

**Comment by wgwz** [comment]
just recording the logs from the `dev` build for the migration:

```
2023-05-05T18:38:57.246535+00:00 app[release.3935]: Current version of schema "public": 2.25
2023-05-05T18:38:57.290099+00:00 app[release.3935]: Migrating schema "public" to version "3.01 - db cleanup"
2023-05-05T18:38:57.390874+00:00 app[release.3935]: DB: drop cascades to 3 other objects
2023-05-05T18:38:57.402897+00:00 app[release.3935]: DB: drop cascades to constraint credit_class_methodology_id_fkey on table credit_class
2023-05-05T18:38:57.414201+00:00 app[release.3935]: DB: drop cascades to constraint transaction_purchase_id_fkey on table transaction
2023-05-05T18:38:57.416601+00:00 app[release.3935]: DB: drop cascades to function retire_credits(uuid,uuid,uuid,numeric,jsonb)
2023-05-05T18:38:57.423442+00:00 app[release.3935]: DB: drop cascades to 3 other objects
2023-05-05T18:38:57.432775+00:00 app[release.3935]: DB: function private.really_create_project() does not exist, skipping
2023-05-05T18:38:57.434086+00:00 app[release.3935]: DB: function private.really_create_user_if_needed() does not exist, skipping
2023-05-05T18:38:57.438380+00:00 app[release.3935]: DB: function public.really_create_user() does not exist, skipping
2023-05-05T18:38:57.438610+00:00 app[release.3935]: DB: function public.really_create_user_if_needed() does not exist, skipping
2023-05-05T18:38:57.439484+00:00 app[release.3935]: DB: function retire_credits() does not exist, skipping
2023-05-05T18:38:57.446887+00:00 app[release.3935]: DB: skipping...
2023-05-05T18:38:57.449618+00:00 app[release.3935]: DB: migrating organization.website to party.website_link...
2023-05-05T18:38:57.488404+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "auth"
2023-05-05T18:38:57.476125+00:00 app[web.1]: server: 2023-05-05T18:38:57.475Z graphile-build-pg Schema change detected: re-inspecting schema...
2023-05-05T18:38:57.579252+00:00 app[web.1]: server: 2023-05-05T18:38:57.579Z graphile-build-pg Schema change detected: re-inspecting schema complete
2023-05-05T18:38:57.520646+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "create org"
2023-05-05T18:38:57.540618+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "create project"
2023-05-05T18:38:57.554115+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "create user"
2023-05-05T18:38:57.571261+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "get available credits"
2023-05-05T18:38:57.586753+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "get contact email"
2023-05-05T18:38:57.606513+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "get current user id"
2023-05-05T18:38:57.618085+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "is admin"
2023-05-05T18:38:57.632562+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "issue credits"
2023-05-05T18:38:57.647082+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "retire credits"
2023-05-05T18:38:57.660310+00:00 app[release.3935]: Migrating schema "public" with repeatable migration "transfer credits"
2023-05-05T18:38:57.671587+00:00 app[release.3935]: Successfully applied 12 migrations to schema "public", now at version v3.01 (execution time 00:00.446s)
2023-05-05T18:38:57.778942+00:00 app[release.3935]: Done in 11.65s.
```

**Pull request merged**